### DISCUSSION TITLE
Consider using a `struct` instead of int in `Commands` tutorial

### DISCUSSION THREAD

**Discussion by erwin** [charmbracelet/bubbletea#562] [Ideas]
In the ["Commands in Bubble Tea" tutorial](https://github.com/charmbracelet/bubbletea/tree/master/tutorials/commands/), the demonstration `checkServer()` function returns:

```
return statusMsg(res.StatusCode)
```

I read that tutorial page a few times, and sadly for me, it wasn't obvious to me that `statusMsg` could be a struct and thus I could easily **asynchronously return multiple values**.

After seeing @bashbunni 's  [video](https://www.youtube.com/watch?v=uJ2egAkSkjg) and more importantly, reviewing the code in the project that goes with the video, I saw:

[pjs/tui/projectui/model.go](https://github.com/bashbunni/pjs/blob/main/tui/projectui/model.go)
```
// SelectMsg the message to change the view to the selected entry
type SelectMsg struct {
	ActiveProjectID uint
}
```

and then later:
[pjs/tui/projectui/commands.go](https://github.com/bashbunni/pjs/blob/main/tui/projectui/commands.go)
```
func selectProjectCmd(ActiveProjectID uint) tea.Cmd {
	return func() tea.Msg {
		return SelectMsg{ActiveProjectID: ActiveProjectID}
	}
}
```

Of course `tea.Msg` is defined in tea.go as: `type Msg interface{}`, so I get it now. 

I realize that you kind of address this in the following section on Cmds "One More Thing About Commands" but even the "a more real-world example" is still returning an int rather than a struct.

```
func checkSomeUrl(url string) tea.Cmd {
    return func() tea.Msg {
        c := &http.Client{Timeout: 10 * time.Second}
        res, err := c.Get(url)
        if err != nil {
            return errMsg{err}
        }
        return statusMsg(res.StatusCode)
    }
}
```

Similar issue in the `return someMsg{id: id}` example... If you had shown the definition of `somemsg` I would have put it together.

```
type someMsg struct {
  id: int
}
```

For some reason, even when I read:

> Cmds are defined in Bubble Tea as type Cmd func() Msg. So they're just functions that don't take any arguments and return a **Msg, which can be any type**

In my mind, I read that as "can be any simple type". It would have helped me if a little more verbose, explicitly using words from go syntax like:

> "is defined as `interface{}` so it can be a composite type like a struct{} or any other type"

Thanks for taking the time to read this. Not trying to criticize. The tutorial is very helpful. Just hopefully some feedback that may make the tutorial even more useful for people coming from other languages.

**Comment by bashbunni** [comment]
Thanks for the feedback! Do you think it would be helpful to maybe amend the commands tutorial to point to some of our examples of different data types being used as `tea.Msg`?

Something like:
> Cmds are defined in Bubble Tea as type Cmd func() Msg. So they're just functions that don't take any arguments and return a Msg, which can be **any** type e.g. [structs](https://github.com/charmbracelet/bubbletea/blob/master/examples/send-msg/main.go), ints, strings, etc.

Where we would include links to examples using these different data types. Let us know your thoughts!

**Reply by erwin** [reply]
Yes.

And two other points to consider pointing out:

1) If you have nested models, you need to pay attention to both the Cmd output of each model and then call `tea.Batch` to process them all... If you ever overlook handling the return values from ANY nested model, you'll probably get behavior that's not what your anticipating.

2) You can have a Cmd message to a Cmd to a Cmd... They can be chained. This is not necessarily ideal practice, but I think it helps elucidate the actual operation in the mind of the developer.

**Reply by bashbunni** [reply]
Ok that makes sense. I'll take another look at the docs and be sure to mention these things in another one of our tutorials on YouTube
